packages/uikit-react-native/src/components/FileViewer/FileViewerContent.tsx

```diff
@@ -0,0 +1,141 @@
+import { ReactNativeZoomableView, ReactNativeZoomableViewProps } from '@openspacelabs/react-native-zoomable-view';
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import { ImageProps, ImageStyle, ImageURISource, StyleProp, StyleSheet, useWindowDimensions } from 'react-native';
+
+import {
+  Box,
+  Image,
+  LoadingSpinner,
+  createStyleSheet,
+  useHeaderStyle,
+  useUIKitTheme,
+} from '@sendbird/uikit-react-native-foundation';
+import { FileType, useIIFE } from '@sendbird/uikit-utils';
+
+import { usePlatformService } from '../../hooks/useContext';
+import SBUUtils from '../../libs/SBUUtils';
+
+type Props = {
+  type: FileType;
+  src: string;
+  topInset?: number;
+  bottomInset?: number;
+  maxZoom?: number;
+  minZoom?: number;
+  onPress?: () => void;
+};
+const FileViewerContent = ({ type, src, topInset = 0, bottomInset = 0, maxZoom = 4, minZoom = 1, onPress }: Props) => {
+  const [loading, setLoading] = useState(true);
+
+  const { defaultHeight } = useHeaderStyle();
+  const { mediaService } = usePlatformService();
+  const { palette } = useUIKitTheme();
+
+  const source = { uri: src };
+  const onLoadEnd = () => setLoading(false);
+  const mediaViewer = useIIFE(() => {
+    switch (type) {
+      case 'image': {
+        return (
+          <ZoomableImageView
+            source={source}
+            style={StyleSheet.absoluteFill}
+            resizeMode={'contain'}
+            onLoadEnd={onLoadEnd}
+            zoomProps={{
+              minZoom,
+              maxZoom,
+              onTouchEnd: onPress,
+            }}
+          />
+        );
+      }
+
+      case 'video':
+      case 'audio': {
+        return (
+          <mediaService.VideoComponent
+            source={source}
+            style={[StyleSheet.absoluteFill, { top: topInset, bottom: defaultHeight + bottomInset }]}
+            resizeMode={'contain'}
+            onLoad={onLoadEnd}
+          />
+        );
+      }
+
+      default: {
+        return null;
+      }
+    }
+  });
+
+  return (
+    <Box style={styles.container}>
+      {mediaViewer}
+      {loading && <LoadingSpinner style={{ position: 'absolute' }} size={40} color={palette.primary300} />}
```

Comment:
```suggestion
      {loading ? <LoadingSpinner style={{ position: 'absolute' }} size={40} color={palette.primary300} /> : mediaViewer}
```
Should we display only one component depends on the loading status? Or it's just okay to keep both like the above?

Reply:
The loading state is controlled by the `onLoadEnd` event of the mediaViewer(ImageView or VideoView).
Therefore, the LoadingSpinner should take the form of covering the screen.